envoy/server/admin.h

```diff
@@ -279,6 +279,11 @@ class Admin {
    */
   static RequestPtr makeStaticTextRequest(absl::string_view response_text, Http::Code code);
   static RequestPtr makeStaticTextRequest(Buffer::Instance& response_text, Http::Code code);
+
+  /**
+   * Closes the listening socket for the admin.
+   */
+  virtual void closeSocket() PURE;
```

Comment:
I can't seem to figure out why this new pure virtual method is needed. Oh is it because we're using `std::unique_ptr<Admin> admin_` instead of `std::unique_ptr<AdminImpl> admin_` in server.h? Do we need to make that change?

Reply:
I believe you, but I don't yet understand how to connect all the dots :) Why does the server need an API pointer now when before it had an impl pointer?